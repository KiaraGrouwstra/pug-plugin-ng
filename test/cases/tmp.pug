ng-container([ngSwitch]='type')
  template([ngSwitchCase]=`'object'`)
    object([val]=`val` [schema]=`new_schema` [named]=`false`)
  template([ngSwitchCase]=`'array'`)
    array([val]=`val` [schema]=`new_schema` [named]=`false`)
  template([ngSwitchCase]=`'scalar'`)
    ng-container([ngSwitch]=`isHtml(val)`)
      template([ngSwitchCase]=`true`)
        myiframe([val]=`val`)
      template(ngSwitchDefault)
        .scalar([innerHtml]=`val | scalar:new_schema`)
  template(ngSwitchDefault)
    p UNIMPLEMENTED OUTPUT-VALUE TYPE ({{ type | json }})!
