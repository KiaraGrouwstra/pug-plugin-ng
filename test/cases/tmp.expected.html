
<ng-container [ngSwitch]="type">
  <template [ngSwitchCase]="'object'">
    <object [val]="val" [schema]="new_schema" [named]="false"></object>
  </template>
  <template [ngSwitchCase]="'array'">
    <array [val]="val" [schema]="new_schema" [named]="false"></array>
  </template>
  <template [ngSwitchCase]="'scalar'">
    <ng-container [ngSwitch]="isHtml(val)">
      <template [ngSwitchCase]="true">
        <myiframe [val]="val"></myiframe>
      </template>
      <template ngSwitchDefault>
        <div class="scalar" [innerHtml]="val | scalar:new_schema"></div>
      </template>
    </ng-container>
  </template>
  <template ngSwitchDefault>
    <p>UNIMPLEMENTED OUTPUT-VALUE TYPE ({{ type | json }})!</p>
  </template>
</ng-container>